content/zh/docs/tasks/administer-cluster/kms-provider.md

```diff
@@ -0,0 +1,283 @@
+
+<!-- * Using languages other than Go: Use the protoc compiler with the proto file: [service.proto](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/v1beta1/service.proto) to generate a stub file for the specific language -->
+
+* 使用 Go 以外的其他语言：用 protoc 编译器编译 proto 文件： [service.proto](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/v1beta1/service.proto) 为指定语言生成存根文件。
+
+<!-- Then use the functions and data structures in the stub file to develop the server code. -->
+
+然后使用存根文件中的函数和数据结构开发服务器代码。
+
+<!-- **Notes:** -->
+
+**注意：**
+
+<!-- * kms plugin version: `v1beta1`
+
+In response to procedure call Version, a compatible KMS plugin should return v1beta1 as VersionResponse.version -->
+
+* kms 插件版本：`v1beta1`
+
+作为对过程调用 Version 的响应，兼容的 KMS 插件应把 v1beta1 作为 VersionResponse.version 返回
+
+<!-- * message version: `v1beta1`
+
+All messages from KMS provider have the version field set to current version v1beta1 -->
+
+* 消息版本：`v1beta1`
+
+所有来自 KMS 提供商的消息都把 version 字段设置为当前版本 v1beta1
+
+<!-- * protocol: UNIX domain socket (`unix`)
+
+The gRPC server should listen at UNIX domain socket -->
+
+* 协议：UNIX 域套接字 (`unix`)
+
+gRPC 服务器应监听 UNIX 域套接字
+
+<!-- ### Integrating a KMS plugin with the remote KMS
+The KMS plugin can communicate with the remote KMS using any protocol supported by the KMS.
+All configuration data, including authentication credentials the KMS plugin uses to communicate with the remote KMS, 
+are stored and managed by the KMS plugin independently. The KMS plugin can encode the ciphertext with additional metadata that may be required before sending it to the KMS for decryption. -->
+
+### 将 KMS 插件与远程 KMS 整合
+KMS 插件可以用任何受 KMS 支持的协议与远程 KMS 通信。
+所有的配置数据，包括 KMS 插件用于与远程 KMS 通信的认证凭据，都由 KMS 插件独立地存储和管理。KMS 插件可以用额外的元数据对密文进行编码，这些元数据是在把它发往 KMS 进行解密之前可能要用到的。
+
+<!-- ### Deploying the KMS plugin 
+Ensure that the KMS plugin runs on the same host(s) as the Kubernetes master(s). -->
+
+### 部署 KMS 插件
+确保 KMS 插件与 Kubernetes 主服务器运行在同一主机上。
+
+<!-- ## Encrypting your data with the KMS provider -->
+## 使用 KMS 提供商加密数据
+
+<!-- To encrypt the data: -->
+为了加密数据：
+
+<!-- 1. Create a new encryption configuration file using the appropriate properties for the `kms` provider: -->
+1. 使用 `kms` 提供商的相应的属性创建一个新的加密配置文件：
+
+```yaml
+kind: EncryptionConfiguration
+apiVersion: apiserver.config.k8s.io/v1
+resources:
+  - resources:
+    - secrets
+    providers:
+    - kms:
+        name: myKmsPlugin
+        endpoint: unix:///tmp/socketfile.sock
+        cachesize: 100
+        timeout: 3s
+    - identity: {}
+```
+
+<!-- 2. Set the `--encryption-provider-config` flag on the kube-apiserver to point to the location of the configuration file. -->
+2. 设置 kube-apiserver 的 `--encryption-provider-config` 参数指向配置文件的位置。
+<!-- 3. Restart your API server. -->
```

Comment:
I have bad experience before of adding comments in between the bullets (i.e. the number wont be reused as 1., 2., 3. in the sequential way), so if the preview does not good (i have not checked yet), i highly recommend to use the block comment out of the original english text so that the numbered items would be consistent.

Reply:
有序编号列表 在有 yaml 插入时会乱掉，甚至英文版也是如此。参见 https://kubernetes.io/docs/tasks/administer-cluster/kms-provider/。我尝试过多种方法都无法使有序列表正常，看起来似乎是 hugo 的问题。

PS:release-1.14 下我使用的是 hugo-0.54.0；hugo-0.55.6 编译失败